plugins/outline.py

```diff
@@ -0,0 +1,311 @@
+"""
+Representation Outliner
+https://pymolwiki.org/index.php/Outline
+Author: Jarrett Johnson (Schrodinger, Inc.)
+"""
+
+from pymol import cmd
+from pymol.Qt import QtCore
+from pymol.Qt import QtGui
+from pymol.Qt import QtWidgets
+
+import os
+from io import BytesIO
+
+from PIL import Image
+from PIL import ImageChops
+from PIL import ImageDraw
+from PIL import ImageFilter
+from PIL import ImageOps
+
+VERSION = "0.1"
+
+
+def __init_plugin__(app=None) -> None:
+    from pymol.plugins import addmenuitemqt
+    addmenuitemqt('Outliner', run_plugin_gui)
+
+
+def run_plugin_gui() -> None:
+    '''
+    Open our custom dialog
+    '''
+    global dialog
+
+    if dialog is None:
+        dialog = make_dialog()
+
+    dialog.show()
+
+
+# global reference to avoid garbage collection
+dialog = None
+
+
+def _create_clean_overlay(img: Image, target_color: tuple) -> Image:
+    """
+    Create a clean overlay of the image by removing inner edges
+    :param img: Image
+    :param target_color: Outline color
+    """
+    BLACK = (0, 0, 0, 255)
+    WHITE = (255, 255, 255, 255)
+    TRANSPARENT = (0, 0, 0, 0)
+
+    # Convert to grayscale
+    img = img.convert('L')
+
+    # Floodfill approach adopted from
+    # https://learnopencv.com/filling-holes-in-an-image-using-opencv-python-c/
+
+    # Threshold
+    threshold_val = 255 / 2
+    threshold = img.point(lambda p: 255 if p > threshold_val else 0)
+    threshold = threshold.convert("RGB")
+
+    # Inverted threshold
+    inverted = threshold.copy()
+    seed = (0, 0)
+    ImageDraw.floodfill(inverted, seed, WHITE[0:3], thresh=200)
+    inverted = ImageOps.invert(inverted)
+
+    # Combine
+    bitwise_or_add = ImageChops.add(inverted, threshold)
+
+    # Find edges
+    edges = bitwise_or_add.filter(ImageFilter.FIND_EDGES).convert("RGBA")
+
+    def _replace_color(img: Image, target_color: tuple,
+                       replace_color: tuple) -> None:
+        """
+        Replace color in image
+        :param img: Image
+        :param target_color: Color to replace
+        :param replace_color: Color to replace with
+        """
+        data = img.getdata()
+        img.putdata(
+            [item if item != target_color else replace_color for item in data])
+
+    _replace_color(edges, BLACK, TRANSPARENT)
+    _replace_color(edges, WHITE, target_color)
```

Comment:
> Very nice. Works as advertised. I only have a few minor comments.
> 
> Any chance to get antialiasing for the outline? I think it would be possible with oversampling.
> 
> Line width could be controlled with [PIL.ImageFilter.MaxFilter](https://pillow.readthedocs.io/en/stable/reference/ImageFilter.html#PIL.ImageFilter.MaxFilter)

Good suggestions. I will try tonight.